### PR TITLE
Redesign of the app

### PR DESCRIPTION
**Pull request by dsavenko**
Hi,

This PRs is a redesign of the app. With it, the app becomes a mere wrapper for command-line 'classification'-type programs. It becomes far more simple and easy to maintain. Almost all CK-related logic are extracted out of the app (utilising existing CK capabilities), so adding new engines/models/datasets can be done without modifying the app.

It also fixes problems (notably on Windows), which are depicted here: https://github.com/dividiti/crowdsource-video-experiments-on-desktop/issues?q=is%3Aopen+is%3Aissue+milestone%3A%22March%2C+1%22 . Let's do some testing, then we can close them.

It may introduce new problems, of course :) But they'll be much easier to fix, since now we're not dealing with tricky stuff like manipulating DLLs at runtime.

Th preprocess script includes a 'ck' binary location discovery mechanism. If it works fine, we can include it into the CK kernel (consider this as a testing ground for it).

The way you supposed to run the app is also changed - please, take a look at README.

### CODE REVIEW

**Comment by dsavenko** [comment]
I checked that this PR fixes the following issues: https://github.com/dividiti/crowdsource-video-experiments-on-desktop/issues/6, https://github.com/dividiti/crowdsource-video-experiments-on-desktop/issues/14, https://github.com/dividiti/crowdsource-video-experiments-on-desktop/issues/17, https://github.com/dividiti/crowdsource-video-experiments-on-desktop/issues/20, https://github.com/dividiti/crowdsource-video-experiments-on-desktop/issues/23

File: program/dnn-desktop-demo/preprocess.py [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+class WhichError(Exception):
+    pass
+
+#---- internal support stuff
+
+def _getRegisteredExecutable(exeName):
+    """Windows allow application paths to be registered in the registry."""
+    registered = None
+    if sys.platform.startswith('win'):
+        if os.path.splitext(exeName)[1].lower() != '.exe':
```
Comment (DVEfremov) [inline program/dnn-desktop-demo/preprocess.py:180]: Could be index [1] out of range after split ?

File: program/dnn-desktop-demo/preprocess.py [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+            exeName += '.exe'
+        import _winreg
+        try:
+            key = "SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\App Paths\\" +\
+                  exeName
+            value = _winreg.QueryValue(_winreg.HKEY_LOCAL_MACHINE, key)
+            registered = (value, "from HKLM\\"+key)
```
Comment (DVEfremov) [inline program/dnn-desktop-demo/preprocess.py:187]: HKLM -> HTLM ?

Comment (DVEfremov) [reply program/dnn-desktop-demo/preprocess.py:187]: looks like it's HKEY LOCAL MACHNE )

**Review by DVEfremov — COMMENTED** [review]
Looks good for me

File: program/dnn-desktop-demo/preprocess.py (hunk 1 above)
Comment (dsavenko) [reply program/dnn-desktop-demo/preprocess.py:180]: No, that's not possible. See the docs about `os.path.splitext`

File: program/dnn-desktop-demo/preprocess.py (hunk 2 above)
Comment (dsavenko) [reply program/dnn-desktop-demo/preprocess.py:187]: Yeah, Windows registry key :)

File: program/dnn-desktop-demo/preprocess.py (hunk 1 above)
Comment (DVEfremov) [reply program/dnn-desktop-demo/preprocess.py:180]: ok, thanks

**Comment by DVEfremov** [comment]
I think this PR could be merged

**Pull request merged**